### DISCUSSION TITLE
Dynamic key

### DISCUSSION THREAD

**Discussion by pnkapadia64** [MichalLytek/type-graphql#1081] [General]
I'm creating a schema for a game that has a translation map for its name. The map could have one of the locale's as its key and the name as its value.
```
name: {
    [key: string]: string;
}

Example
name: {
    en_US: 'Game',
    es_ES: 'Juego'
}
```
How do we define this field of name in type-graphql?

**Comment by MichalLytek** [comment]
It's a general GraphQL question about "dynamic" keys:

```graphql
enum Locale {
  en_US
  es_ES
}

type Translation {
  key: Locale!
  value: String!
} 

type Foo {
  id: Int!
  name: [Translation!]!
}
```
